cylc/flow/tests/test_rundb.py

```diff
@@ -170,5 +196,76 @@ def test_upgrade_hold_swap():
         assert not dao.upgrade_is_held()
 
 
+def test_upgrade_to_platforms(tmp_path):
+    """Test upgrader logic for platforms in the database.
+    """
+    # Set up the globalrc
+    set_up_globalrc(GLOBALRC, tmp_path)
```

Comment:
Where does `tmp_path` come from?

For bonus marks you could use a context manager something like this:

```python
with set_up_globalrc(GLOBALRC):
    # the temp file has been written
# the temp file has been removed
```

Reply:
`tmp_path` is a pytest fixture - it's just something that Pytest magically gives you see https://docs.pytest.org/en/latest/tmpdir.html

Reply:
Cool, learn something new every day. So long a pytest cleans up `tnp_path` afterwards that's great.